clang/lib/3C/ProgramInfo.cpp

```diff
@@ -1047,8 +1047,8 @@ void ProgramInfo::addTypedef(PersistentSourceLoc PSL, bool ShouldCheck,
   auto Name = "typedef__" + TD->getNameAsString();
   auto* PV = new PointerVariableConstraint(TD->getUnderlyingType(), nullptr,
                                        Name, *this, C);
-  if (ShouldCheck && canWrite(PSL.getFileName()))
-    this->typedefVars[PSL] = {*PV};
-   else
-    this->typedefVars[PSL] = {};
+  if (!(ShouldCheck && canWrite(PSL.getFileName())))
+    PV->constrainToWild(this->getConstraints(), "wild", &PSL);
```

Comment:
Thanks for adding a root cause of wildness; I meant to ask you about this.

Please add different, more descriptive messages for `ShouldCheck` (I think this has something to do with inline structs? Maybe this is a good opportunity to make the parameter name more specific so we realize if we introduce a new root cause in the future that needs a different message) and `canWrite` (this can be "Declaration in non-writable file" like the others) and tests for them in `clang/test/3C/root_cause.c`.

Reply:
Oh yes. You're right. Missed that after testing.

Reply:
And the tests in `clang/test/3C/root_cause.c` please?